pkg/apis/kf/v1alpha1/source_lifecycle.go

```diff
@@ -74,6 +74,7 @@ func (status *SourceStatus) PropagateBuildStatus(build *build.Build) {
 	}
 
 	status.BuildName = build.Name
+	status.manage().MarkUnknown(SourceConditionBuildSucceeded, "initializing", "Build in progress")
```

Comment:
Could we clean up build after it finishes the build-image and then sync source status?

Reply:
I don't think we want to clean them up right away, however I agree we need to GC them at some point. I suspect (but I've not looked into it), that builds in Tekton and Knative Build are GC'd at some point (similar to revisions in Knative serving).

We should ensure that our source controller doesn't bring a GC'd build back and that our sources are GC'd as well: #459